core-rust/state-manager/src/mempool/pending_transaction_result_cache.rs

```diff
@@ -510,10 +512,10 @@ impl PendingTransactionResultCache {
         let mut write_mempool = self.mempool.write();
         if attempt.rejection.is_some() {
             write_mempool.remove_by_payload_hash(&notarized_transaction_hash);
-        } else if let Some(state_version) = attempt.against_state.specific_version() {
+        } else if let AtState::Specific(specific_state) = &attempt.against_state {
             write_mempool.update_transaction_executed_state_version(
                 &notarized_transaction_hash,
-                state_version,
+                specific_state.committed_version(),
```

Comment:
Optional thought - on reflection, I think this would be clearer as `mempool.observe_pending_execution_result(&notarized_transaction_hash, &attempt)` - and then the logic for "what should I do with this knowledge" moves into the mempool 👍